pkg/tsdb/postgres/postgres.go

```diff
@@ -56,15 +57,124 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	return endpoint, err
 }
 
+func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
+	logger := log.New("tsdb.postgres")
+	logger.Debug("Creating Postgres query endpoint")
+	endpoint, err := databaseConnection(datasource, logger)
+	return endpoint, err
+}
+
 // escape single quotes and backslashes in Postgres connection string parameters.
 func escape(input string) string {
 	return strings.ReplaceAll(strings.ReplaceAll(input, `\`, `\\`), "'", `\'`)
 }
 
+func writeConnectionFile(ds *models.DataSource, fileContent string, currentPath string, certFileName string,
+	jsonFieldName string) error {
+	var generatedFilePath string
+	if tlsjs, ok := ds.JsonData.CheckGet(jsonFieldName); ok {
+		generatedFilePath = tlsjs.MustString("")
+	}
+	if fileContent != "" {
+		if generatedFilePath == "" {
+			generatedFilePath = filepath.Join(currentPath, certFileName)
+		}
+
+		if err := ioutil.WriteFile(generatedFilePath, []byte(fileContent), 0700); err != nil {
```

Comment:
This file should definitely not be executable.
```suggestion
		if err := ioutil.WriteFile(generatedFilePath, []byte(fileContent), 0600); err != nil {
```